### PR TITLE
Apply impulse based on input_state. Also apply drag.

### PR DESCRIPTION
**Pull request by robertkarl**
# Overview

Goal: imitate the physics of a mid-sized agar body.
To determine velocity each frame:
- Always apply a force based on the state of the WASD keys
- Apply drag (opposite velocity vector with magnitude `k * speed`

After these two steps clamp velocity to be at most magnitude `this.speed`

Body skids around a bit, accelerates relatively quickly to max and doesn't go much beyond that.
# Test plan

bump into walls, other bodies, try to exceed max speed
# TODO

Do we have an assert/error function? otherwise remove a log statement

### CODE REVIEW

**Comment by robertkarl** [comment]
Updated. Set max velocity, add drag

File: PlayerMovementComponent.js [hunk 1]
```diff
@@ -1,8 +1,14 @@
 var Box2D = require('box2dweb');
+var clamp = require('./clamp');
 var b2Vec2 = Box2D.Common.Math.b2Vec2;
 
 var EntityComponent = require('./EntityComponent');
 
+// Proportional to surface drag
+var drag_constant = 0.05;
```
Comment (zackgomez) [inline PlayerMovementComponent.js]: consts shoudl be declared as

const ALL_CAPS_NAME = derp;

File: PlayerMovementComponent.js [hunk 2]
```diff
@@ -24,6 +31,41 @@ class PlayerMovementComponent extends EntityComponent {
...
+    if (input_state.forward) {
+      yvel_target += -1;
+    }
+    if (input_state.backward) {
+      yvel_target += 1;
+    }
+    var impulse = b2Vec2.Make(xvel_target * MOVEMENT_IMPULSE_CONSTANT,
+          yvel_target * MOVEMENT_IMPULSE_CONSTANT);
+
+    body.ApplyImpulse(impulse, body.GetWorldCenter());
```
Comment (zackgomez) [inline PlayerMovementComponent.js]: nit: dont apply 0 impulses

Comment (robertkarl) [inline PlayerMovementComponent.js]: Skipped applying 0 impulses in two places

**Comment by robertkarl** [comment]
Delete invariant use

File: PlayerMovementComponent.js [hunk 3]
```diff
@@ -24,6 +30,45 @@ class PlayerMovementComponent extends EntityComponent {
...
+      yvel_target += 1;
+    }
+    var impulse = b2Vec2.Make(xvel_target * MOVEMENT_IMPULSE_CONSTANT,
+          yvel_target * MOVEMENT_IMPULSE_CONSTANT);
+    if (impulse.Length()) {
+      body.ApplyImpulse(impulse, body.GetWorldCenter());
+    }
+
+
+    if (!impulse.Length()) {
```
Comment (zackgomez) [inline PlayerMovementComponent.js]: lol 'else'

**Comment by robertkarl** [comment]
Use this cool language feature called `else`

**Comment by robertkarl** [comment]
Update formatting.

**Pull request merged**